File: tests/fixtures/swift-deadcode-test/Sources/ComplexReferences.swift
/// 測試複雜引用情況
import Foundation

// MARK: - 非 DEADCODE: 被 processItems 使用

/// Transform a single item
func transformItem(_ item: String) -> String {
    return item.uppercased()
}

// MARK: - 非 DEADCODE: 被外部使用

/// Process a list of items
func processItems(_ items: [String]) -> [String] {
    return items.map { transformItem($0) }
}

// MARK: - DEADCODE: 定義了但沒有使用

/// Unused transformer function
func unusedTransformer(_ value: Int) -> Int {
    return value * 2
}

// MARK: - 非 DEADCODE: 被 createProcessor 使用

/// Processor class
class Processor {
    let name: String

    init(name: String) {
        self.name = name
    }

    func run() -> String {
        return "Running \(name)"
    }
}

// MARK: - 非 DEADCODE: 工廠函式

/// Create a processor instance
func createProcessor(name: String) -> Processor {
    return Processor(name: name)
}

// MARK: - DEADCODE: 未使用的工廠函式

/// Create an unused processor
func createUnusedProcessor() -> Processor {
    return Processor(name: "unused")
}

// 使用這些函式
let result = processItems(["a", "b", "c"])
let processor = createProcessor(name: "main")

File: tests/fixtures/python-deadcode-test/complex_references.py
"""
測試複雜引用情況
"""
from typing import List, Optional


# 非 DEADCODE: 被 process_items 使用
def transform_item(item: str) -> str:
    """Transform a single item"""
    return item.upper()


# 非 DEADCODE: 被外部使用
def process_items(items: List[str]) -> List[str]:
    """Process a list of items"""
    return [transform_item(item) for item in items]


# DEADCODE: 定義了但沒有使用
def unused_transformer(value: int) -> int:
    """Unused transformer function"""
    return value * 2


# 非 DEADCODE: 被 create_processor 使用
class Processor:
    """Processor class"""

    def __init__(self, name: str):
        self.name = name

    def run(self):
        return f"Running {self.name}"


# 非 DEADCODE: 工廠函式
def create_processor(name: str) -> Processor:
    """Create a processor instance"""
    return Processor(name)


# DEADCODE: 未使用的工廠函式
def create_unused_processor() -> Processor:
    """Create an unused processor"""
    return Processor("unused")


# 使用這些函式
result = process_items(["a", "b", "c"])
processor = create_processor("main")
